docs/concepts/glossary.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Glossary
+nav-pos: 3
+nav-title: Glossary
+nav-parent_id: concepts
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+#### Flink Application Cluster
+
+A Flink Application Cluster is a dedicated [Flink Cluster](#flink-cluster) that only
+executes a single [Flink Job](#flink-job). The lifetime of the
+[Flink Cluster](#flink-cluster) is bound to the lifetime of the Flink Job. Formerly
+Flink Application Clusters were also known as Flink Clusters in *job mode*. Compare to
+[Flink Session Cluster](#flink-session-cluster).
+
+#### Flink Cluster
+
+A distributed system consisting of (typically) one
+[Flink JobManager Process](#flink-jobmanager-process) and one or more
```

Comment:
I have often called this one "master" process, consisting typically of ResourceManager and one or more JobManagers (one per job) and optionally a dispatcher (for session clusters).

For an application cluster: Maser = RM + JM
For as session cluster: Master = RM + Dispatcher + [JM per job]

Reply:
I am not sure is "master" process is the best name. It is similar to the old JobManager, but the name is now a bit misleading.

Reply:
We discussed calling it "Flink Master Process", but this would introduce a lot of inconsistencies through out the documentation. So, we decided to call it Flink JobManager Process for now and revisit this/change it, when we rework the Deployment section of the documentation.

Reply:
I an skeptical whether terms coined in the glossary ever change easily.

We could also do the following:
  - The per-job component inside the master is actually the "JobMaster" in the code.
  - We could call the process as a while (with RM / Disp. / JobMasters) the "Flink Master Process" and redirect "JobManager" there, mentioning that it was the old term used synonymously.
  - Or we keep the term "JobManager" indefinitely and hope that between JobManager and JobMaster there is not too much confusion.

Reply:
I think, we have three options: 

1. Rename "JobManager Process" to "Flink Master" now in the glossary. This would be in line with the future vision, but will create a lot of inconsistencies (e.g. naming of scripts and commands in the binary distribution), which we can not fix as part of the glossary. Redirect can mitigate this a bit.

2. Stick with "JobManager Process" right now and rename later consistently. 

3. Stick with "JobManager" forever and rename the component to "JobMaster" in accordance to the code. I think, it is too late for this (all FLIP-6 documents talk about "JobManager") and the names would also not be very descriptive and very similar.

I am happy with 1. and 2., but not so much with option 3. 

@tillrohrmann, @StephanEwen: What do you think?

Reply:
I would be in favor of 1. with redirecting deprecated terms to it + creating follow up issues to fix the terminology in the other docs + code.

Reply:
My preference would be option (1) as well.

Reply:
Perfect. I'll update the PR accordingly and create a follow-up ticket.

Reply:
Pushed the change and created https://issues.apache.org/jira/browse/FLINK-12639 .